core/txpool/validation.go

```diff
@@ -215,10 +216,19 @@ func ValidateTransactionWithState(tx *types.Transaction, signer types.Signer, op
 	// For LegacyTx, GasFeeCap and GasPrice are equal, so checking GasTipCap and GasFeeCap is enough
 	var minGasTipCap = opts.State.GetState(systemcontracts.PolicyProxyHash, systemcontracts.GetMinGasTipCapStateHash()).Big()
 	var baseFee = opts.State.GetState(systemcontracts.PolicyProxyHash, systemcontracts.GetBaseFeeStateHash()).Big()
-	if math.BigMin(tx.GasTipCap(), new(big.Int).Sub(tx.GasFeeCap(), baseFee)).Cmp(minGasTipCap) < 0 {
+	effectiveGasTip := math.BigMin(tx.GasTipCap(), new(big.Int).Sub(tx.GasFeeCap(), baseFee))
+	if effectiveGasTip.Cmp(minGasTipCap) < 0 {
 		return fmt.Errorf("%w: policy minGasTipCap needed %v, baseFee needed %v, gasTipCap %v, gasFeeCap %v ",
 			ErrUnderpriced, minGasTipCap, baseFee, tx.GasTipCap(), tx.GasFeeCap())
 	}
+	// Apply policy envelope fee check
+	if antimev.IsEnvelope(tx) {
+		var envelopeFee = opts.State.GetState(systemcontracts.PolicyProxyHash, systemcontracts.GetEnvelopeFeeStateHash()).Big()
+		if effectiveGasTip.Cmp(new(big.Int).Add(minGasTipCap, envelopeFee)) < 0 {
+			return fmt.Errorf("%w: policy envelopeFee needed %v, minGasTipCap needed %v, baseFee needed %v, gasTipCap %v, gasFeeCap %v ",
+				ErrUnderpriced, envelopeFee, minGasTipCap, baseFee, tx.GasTipCap(), tx.GasFeeCap())
+		}
+	}
```

Comment:
In fact, here we treat envelope fee as a part of gas tip. So I'd suggest:
```suggestion
	// Apply policy envelope fee check
	if antimev.IsEnvelope(tx) {
		var envelopeFee = opts.State.GetState(systemcontracts.PolicyProxyHash, systemcontracts.GetEnvelopeFeeStateHash()).Big()
		minGasTipCap.Add(minGasTipCap, envelopeFee)
	}
	if math.BigMin(tx.GasTipCap(), new(big.Int).Sub(tx.GasFeeCap(), baseFee)).Cmp(minGasTipCap) < 0 {
		return fmt.Errorf("%w: policy minGasTipCap (including Envelope fee for Envelopes) needed %v, baseFee needed %v, gasTipCap %v, gasFeeCap %v ",
			ErrUnderpriced, minGasTipCap, baseFee, tx.GasTipCap(), tx.GasFeeCap())
	}
```